### ISSUE TITLE
Ability to update the view from a goroutine

### ISSUE THREAD

**Issue by dpotapov** [rivo/tview#6]
Hi,

I'm running some parallel blocking job (like HTTP call) in a goroutine and would like to update the view once it is completed.
Since the main loop is blocked by `screen.PollEvent()` the screen is not redrawing automatically. Some key/resize event have to trigger it. Another problem with `SetFocus()` - it blocks while polling events. 

Here is example: 
```go
app := tview.NewApplication()
a := tview.NewModal().SetText("Waiting 5 seconds...").AddButtons([]string{"Ok"})
b := tview.NewModal().SetText("Done!").AddButtons([]string{"Exit"})

go func() {
	time.Sleep(5 * time.Second) // or http.Get()
	app.SetRoot(b, false).SetFocus(b).Draw()
}()

if err := app.SetRoot(a, false).SetFocus(a).Run(); err != nil {
	panic(err)
}
```

**Comment by rivo** [comment]
Thanks for the hint. I just pushed a change which will fix your example. I decided not to lock while polling for events. I don't see a reason why this couldn't be removed. I hope your issues are resolved with this now. If not, let me know. (Examples always help.)

I won't merge this pull request then.

By the way, `SetRoot()` implicitly calls `SetFocus()`.

**Comment by rivo** [comment]
Sorry, just saw that your change wasn't a pull request. So all good then.

**Comment by dpotapov** [comment]
Thanks!

**Comment by AllanWang** [comment]
@rivo sorry to bring this back from the dead, but some of your examples call `SetRoot()` as well as `SetFocus()`. See [flexbox](https://github.com/rivo/tview/blob/master/demos/flex/main.go). Maybe you'll want to update those examples?

**Comment by rivo** [comment]
Right. I missed that. Should be fixed with the latest commit (c2f07f95484c7bead2bac61de12030815e2aab58). Thanks.
